### PR TITLE
blockdev_commit_with_ignore:commit with io-error:ignore

### PR DESCRIPTION
**Pull request by aliang123**
Live commit with on-error "ignore"

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:1995494

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_commit_with_ignore.q35: PASS (53.33 s)

**Comment by aliang123** [comment]
@zhencliu @ngu-niny, please help to review, thanks

File: qemu/tests/cfg/blockdev_commit_with_ignore.cfg [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    device_tag = 'data'
+    images += " ${device_tag}"
+    force_create_image_data = yes
+    remove_image_data = yes
+    start_vm = yes
+    kill_vm = yes
+    storage_pools = default
+    storage_type_default = "directory"
+    storage_pool = default
+    image_size_data = 2G
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_with_ignore.cfg:14]: Also add image_format_data = qcow2

File: qemu/tests/cfg/blockdev_commit_with_ignore.cfg [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    pre_command += "losetup `cat ${loopdev}` ${backend_base} &&"
+    pre_command += "pvcreate `cat ${loopdev}` &&"
+    pre_command += "vgcreate ${vgname} `cat ${loopdev}` &&"
+    pre_command += "lvcreate -L 100M -n ${lvname} ${vgname} &&"
+    pre_command += "mkfs -t ext3 /dev/${vgname}/${lvname} && mkdir -p ${mount_dir} &&"
+    pre_command += "mount /dev/${vgname}/${lvname} ${mount_dir}"
+
+    post_command += "umount ${mount_dir} &&"
+    post_command += "lvremove -f /dev/${vgname}/${lvname} &&"
+    post_command += "vgremove -f ${vgname} &&"
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_with_ignore.cfg:42]: fyi, we can use lvm.EmulatedLVM, e.g. 
self._lvm = EmulatedLVM(params, get_data_dir())
self._lvm.setup()
self._lvm.cleanup()

cfg:
     vg_name = vg_test
     lv_name = lv_test
     lv_size = 1G
     block_raw_name = lvm_block       
     emulated_image = images/${block_raw_name}
     emulated_image_size = ${image_size_data1}
     force_remove_image = yes
     remove_emulated_image = yes

But it's OK for me about your implementation.

File: qemu/tests/blockdev_commit_with_ignore.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        snapshot_tags = device_params["snapshot_tags"].split()
+        self.device_node = self.get_node_name(device)
+        device = self.get_node_name(snapshot_tags[-1])
+        arguments = {}
+        arguments.update({"on-error": "ignore"})
+        cmd, arguments = backup_utils.block_commit_qmp_cmd(device, **arguments)
+        timeout = self.params.get("job_timeout", 600)
+        self.main_vm.monitor.cmd(cmd, arguments)
+        job_id = arguments.get("job-id", device)
+        job_error = self.wait_until_block_job_error(job_id, timeout)
```
Comment (zhencliu) [inline qemu/tests/blockdev_commit_with_ignore.py:49]: We may use:
job_error = get_event_by_condition(self.main_vm, 'BLOCK_JOB_ERROR', timeout, device=job_id)

**Comment by ngu-niny** [comment]
1-Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_commit_with_ignore Finshed 2021-08-30 00:26:32 2021-08-30 00:41:10 PASS 877.749788

Acked-by: Nini Gu ngu@redhat.com

**Pull request merged**